esmvalcore/preprocessor/_area.py

```diff
@@ -346,31 +386,39 @@ def extract_shape(cube, shapefile, method='contains', crop=True):
     extract_region : Extract a region from a cube.
 
     """
-    if method not in {'contains', 'representative'}:
-        raise ValueError(
-            "Invalid value for `method`. Choose from 'contains', ",
-            "'representative'.")
 
-    with fiona.open(shapefile) as geometries:
-        if crop:
-            cube = _crop_cube(cube, *geometries.bounds)
-        lon_coord = cube.coord(axis='X')
-        lat_coord = cube.coord(axis='Y')
+    geometries = fiona.open(shapefile)
```

Comment:
Can you please restore the `with` block? That will ensure the shapefile is closed even if an exception is raised before `geometries.close()` is reached.

Reply:
done